asmrun/backtrace_prim.c

```diff
@@ -104,7 +205,7 @@ void caml_stash_backtrace(value exn, uintnat pc, char * sp, char * trapsp)
     if (descr == NULL) return;
     /* store its descriptor in the backtrace buffer */
     if (caml_backtrace_pos >= BACKTRACE_BUFFER_SIZE) return;
-    caml_backtrace_buffer[caml_backtrace_pos++] = (backtrace_slot) descr;
+    caml_backtrace_buffer[caml_backtrace_pos++] = extract_debuginfo(descr);
```

Comment:
Currently debuginfo and frame_descr are conflated in rawbacktrace_slot.
This is a problem now that a single slot can point to a list of debuginfo.

The original patch chooses to expose only debuginfo to OCaml world and to project during backtrace generation.
I will change that so that the projection is done only when reading from a raw_backtrace.